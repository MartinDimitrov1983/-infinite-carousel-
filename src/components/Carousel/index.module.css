.carouselContainer {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  overflow: hidden;
  margin-top: 50px
}
