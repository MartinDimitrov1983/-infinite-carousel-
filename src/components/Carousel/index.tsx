import React from 'react';
import { ImageSlider } from '../';
import { useCarousel } from '../../hooks/useCarousel';
import styles from './index.module.css';

interface CarouselProps {
  images: string[];
  height: number;
}

const Carousel: React.FC<CarouselProps> = ({ images, height }) => {
  const { currentIndex, containerRef } = useCarousel(images);

  return (
    <div
      ref={containerRef}
      className={styles.carouselContainer}
      style={{ height: `${height}px` }}
    >
      <ImageSlider images={images} currentImageIndex={currentIndex} />
    </div>
  );
};

export default Carousel;
