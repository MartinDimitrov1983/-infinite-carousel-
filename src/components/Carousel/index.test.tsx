import React from 'react';
import { render } from '@testing-library/react';

import Carousel from './index';

test('Snapshot Carousel', () => {
  const { asFragment } = render(
    <Carousel
      images={[
        'https://media.istockphoto.com/id/1173544006/photo/winding-road.jpg?s=612x612&w=0&k=20&c=_VMEnB08arEsLnbES0knQUWHPrCD8TQFCy99JC4RZIQ=',
        'https://i.pinimg.com/736x/5b/99/b3/5b99b3974c2d889fc1fb703b473394de.jpg',
      ]}
      height={50}
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
