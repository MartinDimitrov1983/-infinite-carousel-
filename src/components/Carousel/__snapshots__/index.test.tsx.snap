// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Carousel 1`] = `
<DocumentFragment>
  <div
    class="carouselContainer"
    style="height: 50vh;"
  >
    <div
      class="imageWrapper"
    >
      <img
        alt="This is image-0"
        class="img"
        src="https://media.istockphoto.com/id/1173544006/photo/winding-road.jpg?s=612x612&w=0&k=20&c=_VMEnB08arEsLnbES0knQUWHPrCD8TQFCy99JC4RZIQ="
        style="display: inline-block;"
      />
    </div>
  </div>
</DocumentFragment>
`;
